articles/forward-and-backward-chaining-in-ai/index.md

```diff
@@ -0,0 +1,119 @@
+
+### Introduction to the Expert System
+Backward and forward chaining originate from the expert system. The expert system has three main components: user interface, inference engine, and knowledge base. 
+
+These three components play specific roles. High-quality and domain-specific knowledge is stored in the knowledge base. The user interface enables users of the system to interact with the expert system.
+
+The inference engine is a component in which logical rules are applied to the knowledge base to get new information or make a decision. This component progresses in two modes: forward and backward chaining. These types of chaining are employed by the inference engine to deduce information. 
+
+![Expert System](/engineering-education/forward-and-backward-chaining-in-ai/expert-system.jpg)
+[Image Source: Tutorials Point](https://www.tutorialspoint.com/artificial_intelligence/images/expert_system.jpg)
+
+### Forward Chaining
+Forward chaining is a method of reasoning in artificial intelligence in which inference rules are applied to existing data to extract additional data until an endpoint (goal) is achieved. 
+
+In this type of chaining, the inference engine starts by evaluating existing facts, derivations, and conditions before deducing new information. An endpoint (goal) is achieved through the manipulation of knowledge that exists in the knowledge base. 
+
+![Forward Chaining](/engineering-education/forward-and-backward-chaining-in-ai/forward-chaining.jpg)
+
+[Image Source: Tutorials Point](https://www.tutorialspoint.com/artificial_intelligence/images/forward_chaining.jpg)
+
+Forward chaining can be used in planning, monitoring, control, and interpretation applications.
+
+#### Properties of Forward Chaining
+* The process uses a down-up approach (bottom to top).
+* It starts from an initial state and uses facts to make a conclusion.
+* This approach is data-driven.
+* It is employed in expert systems and production rule system.
+
+#### Example of Forward Chaining
+A simple example of forward chaining can be explained in the following sequence.
+
+A
+
+A->B
+
+B
+
+A is the starting point. A->B represents a fact. This fact is used to achieve a decision B. 
+
+A practical example will go as follows;
+
+Tom is running (A)
+
+If a person is running, he will sweat (A->B)
+
+Therefore, Tom is sweating. (B)
+
+#### Advantages
+* It can be used to draw multiple conclusions.
+* It provides a good basis for arriving at conclusions.
+* It is more flexible than backward chaining because it does not have a limitation on the data derived from it.
+
+#### Disadvantages
+* The process of forward chaining may be time-consuming. It may take a lot of time to eliminate and synchronize available data.
+* Unlike backward chaining, the hypothesis for this type of chaining is not very clear. The former uses a goal-driven method that arrives at conclusions efficiently. 

```

Comment:
Define hypothesis. Is it the same as ML? It's not clear.

Reply:
I have modified to enhance clarity.